### ISSUE TITLE
[Feature]: JSONForm options for AddRow (blank form) or UpdateRow (populate from sourcedata), so blank forms can detect field types from source data

### ISSUE THREAD

**Issue by GreenFlux** [appsmithorg/appsmith#21995]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

The JSONForm widget can automatically detect data types from the source data, and then set the form input types accordingly. This only works when the source data includes values though. If you want a blank form and provide an object with blank values, the widget assumes every field should be a string. Then every column has to be manually updated to set other types (number, date), which defeats the purpose of auto-generated forms. 

This request is for a new property on the JSONForm, with options for `AddRow` or `UpdateRow`. 
|||
|-|-|
|`UpdateRow`| _Same as current behavior._ The form is populated from the source data. |
|`AddRow`| Source data is used to determine field types only. Values are not populated in the form. |
|`JS`|expression which returns `AddRow` or `UpdateRow`|

### Why should this be worked on?

The JSONForm widget can already detect the type, but users aren't able to take advantage of this feature when they want a blank form. This would be especially useful for large tables with lots of columns that would otherwise need configured manually. 

Additionally, this would allow using the same JSONForm for adding and for updating rows, if the property has a dynamic JS binding. This would simplify UI building, and could also be used to correct the current issues with the _Generate a page from data_ feature, which is populating all AddRow forms with values from a previous row ([Slack thread](https://theappsmith.slack.com/archives/C040LHZN03V/p1678888628826709)).

**Comment by somangshu** [comment]
@GreenFlux can you please elaborate on how the above properties solve the problem, user will still need to define the types

**Comment by GreenFlux** [comment]
@somangshu , the user would continue to define the data types as they do now, by providing JSON with values. The request here is that we add a new property in the JSONForm widget which tells it to use the source data for determining field types only, and not to use it to populate the form. 

> AddRow | Source data is used to determine field types only. Values are not populated in the form.

**Comment by somangshu** [comment]
@GreenFlux if I am getting this right, this will also the dev to use the same form for add new & edit operation. 

cc @dilippitchika @ashit-rath what do you say? Can you shed some light on the use case if thought of in the initial phase?

**Comment by GreenFlux** [comment]
@somangshu , yes! The same form could easily be used for `addRow` or `updateRow`. Then you could use a JS condition to switch the form mode on the fly. Both versions would use the same source data to get the field types, but one would be blank.

**Comment by Nikhil-Nandagopal** [comment]
I think add / update might be a little specific for the use case but we should definitely just add a property to fillDefaultValues: true/false

**Comment by GreenFlux** [comment]
@Nikhil-Nandagopal , that works for me! The `fillDefaultValues` label would probably be more obvious and need less explanation as well. Either way, it solves the issue and allows devs to use the same form for adds and updates, as long as we also give it a JS binding to switch states programatically.

**Comment by somangshu** [comment]
## Stats
| Stat | Values |
| ------------- | ------------- |
| Reach     | 1669 |
| Effort (months)  | 0.75 |

**Comment by pchambless** [comment]
Are you thinking there will be a switch or some indication in the Form Label to indicate whether user is **Inserting** or **Updating** a Row?

**Comment by GreenFlux** [comment]
@pchambless , yes, a switch with a JS option, so that you can change modes programmatically and use the same form for adds and updates.

**Comment by pchambless** [comment]
Nice!  I've had to build some goofy code that can be error prone to re-use the same form for inserts and updates.  This will be nice.
________________________________
